rest/channels.go

```diff
@@ -56,9 +56,17 @@ func (c *Client) LeaveChannel(channel *models.Channel) error {
 // https://rocket.chat/docs/developer-guides/rest-api/channels/info
 func (c *Client) GetChannelInfo(channel *models.Channel) (*models.Channel, error) {
 	response := new(ChannelResponse)
-	if err := c.Get("channels.info", url.Values{"roomId": []string{channel.ID}}, response); err != nil {
-		return nil, err
+	switch {
+	case channel.Name != "":
```

Comment:
Might be important to note... I think grabbing by name actually requires a special permission.

Reply:
Seems to work for me with just the bot permissions. Although this may be because the bot is part of the room. I'll remove the bot from the room and try. If it does not work I'll update above to only use full name if I'd is not specified. I do think it's important to have both especially considering we link to the the API which allows both inside the comments.

Reply:
Just double checked seems to be no additional perms req to do this via name. ( My bot can currently do this with guest permissions as long as it is currently in the room) do you know the specific extra perm in question?